src/classes/frSubscription.cls

```diff
@@ -0,0 +1,119 @@
+        } catch (DMLException e) {
+            insert new Error__c(Error__c =
+                                'Failed to delete funraise registration for registration id ' + funraiseId +
+                                '. Exception: ' + e.getMessage());
+        }
+    }
+    
+    public void create(Map<String, Object> request) {
+        String funraiseId = String.valueOf(request.get('id'));
+        String name = String.valueOf(request.get('name'));
+        String funraiseSupporterId = String.valueOf(request.get('supporterId'));
+        List<Contact> contacts = [SELECT Id from Contact WHERE fr_Id__c = :funraiseSupporterId];
+        if(contacts.isEmpty()) {
+            insert new Error__c(Error__c =
+                                'Failed to find related record: Supporter with id ' + funraiseSupporterId+ ' for subscription id ' + funraiseId);
+            return;
+        }
+        
+        Subscription__c subscription = new Subscription__c(
+        	fr_ID__c = funraiseId,
+            Name = frUtil.truncateToFieldLength(
+                Subscription__c.Name.getDescribe(), 
+                String.valueOf(request.get('name'))
+            ), 
+            Supporter__c = contacts.get(0).Id
+        );
+        
+        String goalId = String.valueOf(request.get('campaignGoalId'));
+        if (String.isNotBlank(goalId)) {
+            List<Campaign> campaigns = [SELECT Id FROM Campaign WHERE fr_ID__c = :goalId];
+            if (campaigns.size() > 0) {
+                subscription.Campaign_Goal__c = campaigns.get(0).Id;
+            } else {
+                insert new Error__c(Error__c =
+                'Failed to find related record: campaign with id ' + goalId+ ' for subscription id ' + funraiseId);
+            }
+        }
+        
+		//transform other request fields into SF custom fields, these are not applicable
+        request.remove('id');
+        request.remove('name');
+        request.remove('supporterId');
+        request.remove('campaignGoalId');
+        request.remove('deleted');
+        
+        String regex = '([a-z])([A-Z]+)';
+        String replacement = '$1_$2';
+        for(String funraiseFieldName : request.keySet()) {
+            String salesforceFieldName = funraiseFieldName.replaceAll(regex, replacement);
+            salesforceFieldName =+ 'funraise__' + salesforceFieldName.capitalize() + '__c';
+            Schema.SObjectField field = fields.get(salesforceFieldName);
+            frUtil.write(subscription, field, funraiseFieldName, request.get(funraiseFieldName), funraiseId);
+        }
+        
+        try {
+            Database.upsert(subscription, Subscription__c.Field.fr_ID__c, true);
+        } catch (DMLException e) {
+            insert new Error__c(Error__c =
+                                'Failed to insert funraise subscription id ' + funraiseId +
+                                '. Exception: ' + e.getMessage());
```

Comment:
Is the exception going to include the field names? seems like it would be helpful if we were going to troubleshoot any problems that may occur with the `funraiseFieldName` -> `salesforceFieldName` conversion code

Reply:
Good catch.  I just checked to see what would happen if we try to call `sObject.put('invalidField', value)`

It throws a runtime exception, which would be unhandled and would trigger an email off to the managed package contact (I think that's me).  I think that would be the correct behavior as we are not expecting any of the fields to not match here